src/core/namespace.c

```diff
@@ -1246,22 +1266,32 @@ static int make_read_only(const MountEntry *m, char **deny_list, FILE *proc_self
                 flags_mask |= MS_NOSUID;
         }
 
+        if (mount_entry_noexec(m)) {
+                new_flags |= MS_NOEXEC;
+                flags_mask |= MS_NOEXEC;
+                submounts = true;
+        } else if (mount_entry_exec(m)) {
+                new_flags &= ~MS_NOEXEC;
+                flags_mask |= MS_NOEXEC;
+                submounts = true;
+        }
+
         if (flags_mask == 0) /* No Change? */
                 return 0;
 
         /* We generally apply these changes recursively, except for /dev, and the cases we know there's
          * nothing further down.  Set /dev readonly, but not submounts like /dev/shm. Also, we only set the
          * per-mount read-only flag.  We can't set it on the superblock, if we are inside a user namespace
          * and running Linux <= 4.17. */
-        submounts =
+        submounts |=
```

Comment:
bitwise or on a `bool`... confusing even if it migt effectively work

Reply:
fixed